pkg/logs/tailers/file/tailer.go

```diff
@@ -344,7 +344,14 @@ func (t *Tailer) forwardMessages() {
 		origin := message.NewOrigin(t.file.Source.UnderlyingSource())
 		origin.Identifier = identifier
 		origin.Offset = strconv.FormatInt(offset, 10)
-		origin.SetTags(append(append(t.tags, t.tagProvider.GetTags()...), output.ParsingExtra.Tags...))
+		tags := append(t.tags, t.tagProvider.GetTags()...)
+
+		if output.ParsingExtra.IsTruncated && coreConfig.Datadog().GetBool("logs_config.tag_truncated_logs") {
+			tags = append(tags, message.TruncatedTag)
```

Comment:
That this is implemented in a file tailer, would it mean that only messages from files would get this tag, while everything else that get's truncated in a decoder would only have `TRUNCATED` appended to it?

Reply:
Yes that is the current behavior. 
(I'll update the PR description). 

My rationale is: 
1. Nearly all of the problems with truncated logs we encounter come from files
2. The current API for attaching tags is not well designed. IMO this code should live inside the decoder, and not be copy pasted into each file tailer. 

In order to solve `2` we would need to do some refactoring on `message` since currently the origin isn't usually copied out of the decoder step when the message is sent to the pipeline. This refactoring would come with some additional risk/churn. 

I plan to open a backlog card to address this in another change, but if you disagree I can shelve this PR and reprioritize this. I mostly wanted this change to make testing auto multiline detection easier.